python/ray/_raylet.pyx

```diff
@@ -2714,7 +2671,46 @@ def _auto_reconnect(f):
 
 
 cdef class GcsClient:
-    """Cython wrapper class of C++ `ray::gcs::GcsClient`."""
+    """
+    Client to the GCS server. Only contains synchronous methods.
+
+    This class is in transition to use the new C++ GcsClient binding. The old
+    PythonGcsClient binding is not deleted until we are confident that the new
+    binding is stable.
+
+    Defaults to the new binding. If you want to use the old binding, please
+    set the environment variable `RAY_USE_OLD_GCS_CLIENT=1`.
+    """
+
+    cdef object inner  # OldGcsClient or NewGcsClient
+    cdef c_bool use_old_client
+
+    def __cinit__(self, address,
+                  nums_reconnect_retry=None,
+                  cluster_id: str = None):
+        self.use_old_client = os.getenv("RAY_USE_OLD_GCS_CLIENT") == "1"
+        if self.use_old_client:
+            self.inner = OldGcsClient(address, nums_reconnect_retry, cluster_id)
+        else:
+            # nums_reconnect_retry is ignored because now we rely on GcsRpcClient
+            # retry.
+            # TODO: it does not support initial connection when GCS is down. We need to
+            # support it in GcsRpcClient.
```

Comment:
What does this mean?

Reply:
I meant if GCS is already down when a GcsClient is created, instead of assuming HA and infinite retrying, we raise exception. Reworded.